### ISSUE TITLE
8. LLMs to Model Agents & Interactions - orienting

### ISSUE THREAD

**Issue by lkcao** [UChicago-Computational-Content-Analysis/Readings-Responses-2024-Winter#14]
Post questions here for this week's oritenting readings:

Park, Joon Sung, Lindsay Popowski, Carrie Cai, Meredith Ringel Morris, Percy Liang, Michael S. Bernstein. 2022. [“Social Simulacra: Creating Populated Prototypes for Social Computing Systems.”](https://dl.acm.org/doi/abs/10.1145/3526113.3545616) UIST: Proceedings of the 35th Annual ACM Symposium on User Interface Software and Technology.

**Comment by XiaotongCui** [comment]
I have some questions regarding the evaluation section of the model. Firstly, the technical evaluation part seems to resemble a variation of the Turing test. However, I believe the purpose of Social Simulacra is to assist designers in making better judgments through simulated social interactions. Even if participants cannot distinguish whether the posts are generated or real, this does not necessarily mean that the simulation is "directionally correct," which may not necessarily aid designers. As for the designer evaluation part, I also have concerns that a sample of 16 designers may not be very persuasive. However, overall, it is a very interesting study!

**Comment by sborislo** [comment]
I think the approach is neat, though I question (or at least, don't understand) the efficacy of the approach. Although the generation of possible outcomes is useful, not knowing the probability of those outcomes reduces the utility, in my view. I think there is an implicit understanding that anything can go wrong, and people often care _how_ things can go wrong; yet, not knowing the probability that a certain outcome comes to fruition makes it tough to know whether planning for a potential outcome is worth the cost. I don't believe GPT is good at assessing probabilities of social outcomes, so it would probably suggest that any possibility is equally likely to occur. This would also mean further downstream social consequences are next to impossible to assess using Social Simulacra. Nonetheless, for generating ideas, I think it's great.

**Comment by yuzhouw313** [comment]
Regarding to ethics in online discourse formation, Park et al. tries to mitigate the effect of anti-social behaviors in prototyping social systems, arguing that these behaviors not only cause the design to be ineffective but also harm individuals. While I agree with the negative effect of anti-social behaviors on online discourse and the absolute urgency to control such behaviors, I am not sure in the prototyping stage why is generating them based on possibly real cases ineffective in design. Additionally, I wonder if the social simulacra's user outcome generation function could serve as a valuable tool for researchers to counteract anti-social behaviors. This approach seems particularly advantageous given the lower stakes of experimenting within a prototype environment as opposed to navigating real, fully populated online discourse.

**Comment by Audacity88** [comment]
I agree with sborislo in finding this a neat idea, but wondering if it is really useful. Basically, the authors are simulating the same process that happens organically: as users join a subreddit, a community emerges, goals are defined, rules are made to deal with undesirable behavior, etc. But it is an incremental process of gradient descent towards optimal rules, not an a priori setup, and I think this way of doing things makes more sense. Would trying to simulate everything in advance really produce better outcomes than the organic growth process? Also, does anyone really consider themself a “designer” of subreddits? What if you designed a perfect, impossible-to-troll subreddit and nobody came?

**Comment by joylin0209** [comment]
Considering the study's highlighted limitations of social simulacra, what directions should future research take to make these prototypes more realistic and useful in varied social computing settings? For instance, how can we better mirror the real world's diversity and complexity in these models? And how do we ensure they don't inadvertently reinforce negative stereotypes?

**Comment by Marugannwg** [comment]
It is intriguing to see that the LLM agent can predict (with some power) what a user might say and post based on their previous actions...

Well, in the real world, we only have one reality and the outcome is deterministic. Now I'm tasting the idea of **MULTIVERSE** proposed in the paper... (which sounds super costly...)
- Any single simulation does not provide enough insight given the probabilistic feature of every agent/step, we want see more..
- Also, it seems to me that any single agent does not provide meaningful insight, given the numerous possibilities of parallel branching in the multiverse.

So, the entire purpose of the simulation is to see some *macroscopic trend* that would happen at a population level. And I wonder what insight it can provide to social scientists if we were to evaluate multiple parallel possibilities. How to ensure the validity there? how can you guarantee you discover anything considering the cost?

**Comment by YucanLei** [comment]
The way they try to simulate an online community is quite stunning. But, how do biases inherent in the large language models used to generate social simulacra impact the diversity and inclusivity of the simulated social interactions, and what steps can be taken to mitigate these biases in the design process?

**Comment by bucketteOfIvy** [comment]
Park et al. (2022) present a novel usage of LLM's that could be of great use for designers attempting to develop online spaces. While I think their evaluation is sufficient for the purposes of the paper -- at the end of the day the model just needs to do good enough that humans can't tell the difference -- I share @XiaotongCui 's concern that the evaluation is probably not as strong as it could be. In particular, if we want to apply community generation methods to studies of communities in general, then it feels like we need to show that not only is there is no human detectable difference between model generated interactions and a random sample of real interactions, but also that more advanced methods are bad at differentiating the two. Is this a fair critique of the evaluation method and, if so, are there common methods in use (e.g. maybe supervised models?) for doing as such?

**Comment by yunfeiavawang** [comment]
This is an impressive paper applying large language models to simulate social systems. The social simulacra technique could be inspirational for system designers to forecast what is going to happen when the population grows in this system. In the evaluation part, the model is justified by the fact that people can not tell apart the real community and the simulated one. But is it enough? To validate the utility of social simulacra, I believe we should take a step further to see if it is forecasting new trends and is susceptible to the designers' manipulations, which is more informative for facilitating social computing design.

**Comment by QIXIN-LIN** [comment]
Fascinating study! I'm curious whether all interactions are solely derived from the training data. Is it conceivable that it might not replicate certain social interactions not covered by the data? For instance, as time progresses, new technologies and other factors could emerge and affect social interactions. Is it possible to identify and follow a fundamental mechanism (or extract it from the training data)?

**Comment by icarlous** [comment]
The simulation of an online community is impressive, but how do the inherent biases in the extensive language models utilized to create social simulations affect the diversity and inclusivity of the simulated social interactions?

**Comment by Caojie2001** [comment]
It's a really interesting article that applies the LLM to simulate interaction on a social media platform. Although the author mentioned in the limitation part that social simulacra will not predict the future, I think this approach still has a promising prospect of application, such as assisting the management of online social platforms. I wonder whether the differences in the online community structure (such as different forms of available interaction) would lead to the differences in interaction patterns and community atmosphere.

**Comment by donatellafelice** [comment]
i wonder about training for other high risk, or dangerous situations, as well as the risks associated with the simulations that are mentioned by the article. the author's write: "social simulacra offers designers a tool for testing their intuitions about the breadth of possible social behaviors that may populate their community, from model citizen behaviors to various edge cases that ultimately become the cracks that collapse a community." while they note that it is text only and soon we may be able to have video etc, would it be possible to use these types of models to create audio situations that should be rehearsed or trained for? would there be a way to move from only text based (like in online posts) to what people might actually say given how many high risk situations are now recorded (ie - 911 calls, hostage negotiations, etc)

**Comment by erikaz1** [comment]
Park et al. (2022) makes a convincing argument that Social simulacra benefits prototyping practices in social computing contexts by providing visual insight (e.g. SimReddit) into the types of content that may be generated within a virtual social space, which designers can use  to improve interactive systems. However, I’m not convinced that the technology relieves all of the subreddit designers’ concerns about the designing process. 

One of my questions is: How is the best set of conversations selected? What differentiates universes in the multiverse feature? If it’s not immediately made clear what the differences between these conversations are, wouldn’t users have to always run the multiverse function (along with always running generate and/or whatif) in order to see a more complete picture/more insight? Additionally, how can we gauge the degrees of differentiation?

**Comment by ethanjkoz** [comment]
Given my interest in social media and online spaces, I found this a very interesting read. Particularly, I enjoyed the section of the paper where the authors discuss the application of WHATIF and GENERATE. However, I am still a little confused on the difference between the two. Also similar to Dan's post, perhaps due to my lack of creativity, I am unsure where SIMREDDIT would benefit actual subreddit creators? (Also I found the concluding remarks quite humorous.)

**Comment by anzhichen1999** [comment]
How might the integration of real-time emotional analysis AI improve the accuracy of social simulacra in predicting complex social dynamics and user interactions in virtual environments

**Comment by alejandro-sarria-morales** [comment]
How do social simulacra account for and adapt for unpredictable social behaviors, particularly in scenarios not directly reflected in the training data of large language models used to simulate these interactions?

**Comment by naivetoad** [comment]
The paper contributes novel techniques for generating social behaviors using GPT-3. What are the specific technical challenges faced when developing these techniques, and how do they compare with traditional methods of prototyping social computing systems?

**Comment by runlinw0525** [comment]
This is a very interesting paper. Here is my question: What are the potential limitations and ethical considerations associated with using social simulacra for prototyping social computing systems, particularly in terms of accurately representing diverse user behaviors and ensuring privacy and consent?

**Comment by muhua-h** [comment]
The motivation behind this work is to demonstrate how to simulate human society. However, due to value alignment trainings, LLMs tend to be very friendly and very prosocial, which is not an accurate reflection of the society. How could we simulate a society with diversity agents in terms of their traits?

**Comment by maxzhuyt** [comment]
"Social Simulacra" is an interesting and beautifully written paper. It seems that the possible simulations we can run incorporate behaviors that are common on the internet and relatively expected (like trolling). However, oftentimes designers may not foresee the creative or subversive usage of their platforms–for example, staging online protests on Github. I wonder what insights we can take from this example to infer more creative user behavior.

**Comment by chenyt16** [comment]
As a psychology student, I know that it is actually very difficult to determine which individual characteristics are responsible for a particular behavior. Therefore, when inputting Persona, how can we determine the information included in the Persona is relevant to behavior?

**Comment by Twilight233333** [comment]
Rather than predictive techniques, I'm more curious about the role of topics that are blocked in real life because of their sensitivity, as authors try to generate responses.

**Comment by HamsterradYC** [comment]
While there are  limits to the complexity or size of the community social simulacra Social Simulacra demonstrates proficiency in simulating community behaviors within certain technical constraints, I‘m curious that with recent technological advancements, could it potentially simulate individual decision-making within complex urban systems, and predict spatial patterns and spillover effects?

**Comment by volt-1** [comment]
Can social simulacra be effectively utilized for crisis management simulations or promoting social good, such as in public health campaigns or emergency response scenarios?

**Comment by yueqil2** [comment]
Can the social simulacra be used in political action considerations, such as military exercises and negotiation simulation?

**Comment by JessicaCaishanghai** [comment]
This is a great reading. How does the concept of "social simulacra" address the limitations of current prototyping practices in social computing, and what potential benefits does it offer for designers in terms of understanding and refining social system behavior at scale?

**Comment by cty20010831** [comment]
I think this paper highlights one important role of LLM (for simulating social behavior) to social science research. But still, I am concerned about what types of prompting techniques are needed in order to mimic the complex social rules (i.e.,, generating diverse user personas and interactions)?

**Comment by floriatea** [comment]
How might the use of social simulacra in the early stages of community design influence the long-term evolution of community norms and behaviors? Can early exposure to simulated behaviors shape moderator and member responses to real-world scenarios in unexpected ways?

**Comment by beilrz** [comment]
I think this is a very interesting paper, the question I had was is it possible to deploy such agent in the wild, and interact with environment passively, in order to simulate human behavior? for example, the agent could process information regarding a social event (such as protest), and decide whether to participate or not. If they decide to participate, they can proceed to absorb the information related to the protest, as if they are there.

**Comment by Dededon** [comment]
I'm interested in how can we tune for "personalities" of AIs with limited input-output tokens. The current GPT-4 model seems to be very "forgetful", and will lose identity after long chains of talk.

**Comment by Vindmn1234** [comment]
Considering the efficacy of social simulacra in generating realistic social interactions and their potential to anticipate the challenges of scaling social computing systems, how do you envision the future of social computing system design evolving with the integration of such advanced prototyping techniques?

**Comment by Brian-W00** [comment]
How might these simulacra adapt to the evolving nature of online discourse, where new slang, cultural references, and social norms emerge rapidly? Could the system effectively update or learn from real-time interactions to stay relevant and provide accurate simulations?

**Comment by Carolineyx** [comment]
Considering the reliance on large language models like GPT-3 to simulate these interactions, how do the authors address the challenges of ensuring diversity and inclusivity within the generated interactions? Moreover, given the dynamic and evolving nature of social norms and behaviors, how can 'Social Simulacra' adapt to reflect these changes over time to continue providing relevant and insightful simulations for future social computing designs?
